### PR TITLE
Enable clangtidies

### PR DESCRIPTION
**Pull request by rschu1ze**
PR #36911 "inverted" ClickHouse's .clang-tidy file. This made it more obvious which checks are not yet enabled (besides less maintenance during compiler upgrades).

I did a test push with all exclusions in .clang-tidy removed, i.e. all checks enabled. Based on the build log, I activated 32 further checks from different categories which either
- are useful and had zero findings in CHs codebase (best case) or
- which provide some benefit and needed limited work to fix them.

Controversial or stylistic checks were omitted. Useful checks which require a ton of changes were also omitted.

Reviewers may take a look the individual commits which include a short description of the check + required changes for each check.

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by rschu1ze** [comment]
@mergifyio update

**Review by devcrafter — COMMENTED** [review]
I don't know all checks but overall LGTM

File: src/Storages/MergeTree/MergeTreeData.cpp [hunk 1]
```diff
@@ -6490,153 +6490,153 @@ ReservationPtr MergeTreeData::balancedReservation(
...
-        std::map<String, std::vector<String>> disk_parts_for_logging;
-        for (const auto & disk : disks)
-            disk_occupation.emplace(disk->getName(), 0);
-
-        std::set<String> committed_big_parts_from_partition;
-        std::set<String> submerging_big_parts_from_partition;
-        std::lock_guard lock(currently_submerging_emerging_mutex);
-
-        for (const auto & part : currently_submerging_big_parts)
+        try
```
[the remark this answers is not in the document]
Comment (rschu1ze) [reply src/Storages/MergeTree/MergeTreeData.cpp:6493]: Sure, can do.

**Comment by rschu1ze** [comment]
Some checks are useful, others are quite obscure, but in any case: reducing the number of exceptions will bring the codebase closer to upstream best practices.

**Review by devcrafter — APPROVED** [review]
LGTM

**Pull request merged**